types/api/v1beta1/event_data.proto

```diff
@@ -0,0 +1,145 @@
+syntax = "proto3";
+
+option go_package = "github.co/aquasecurity/tracee/api/v1beta1";
+
+package tracee.v1beta1;
+
+import "google/protobuf/wrappers.proto";
+
+message EventValue {
+    oneof value {
+        google.protobuf.Int32Value int32 = 1; // intT
+        google.protobuf.Int64Value int64 = 2; // longT
+        google.protobuf.UInt32Value uint32 = 3; // uintT, modeT, devT
+        google.protobuf.UInt64Value uint64 = 4; // ulongT, offT, sizeT, pointerT
+        google.protobuf.StringValue string = 5; // strT
+        google.protobuf.BytesValue bytes = 6; // bytesT
+        google.protobuf.BoolValue bool = 7; // boolT
+        UInt8Value uint8 = 8; // u8T
+        UInt16Value uint16 = 9; // u16T
+        StringArrayValue string_array = 10; // strArrT, argsArrT
+        IntArrayValue int_array = 11; // intArr2T
+        UInt64ArrayValue uint64_array = 12; // uint64ArrT
+        SockAddrValue sockaddr = 13; // sockaddrT
+        CredValue cred = 14; // credT
+        TimespecValue timespec = 15; // timespecT
+        SyscallArgsValue syscall_args = 16;
+        TriggerEvent trigger_event = 17;
+    }
+}
+
+message TriggerEvent {
+    uint32 id = 1;
+    string name = 2;
+    // TODO: is this correct?
+    map<string, EventValue> data = 3;
+}
+
+message StringArrayValue {
+    repeated google.protobuf.StringValue array = 1;
+}
+
+message IntArrayValue {
+    repeated google.protobuf.Int32Value array = 1;
+}
+	
+message UInt64ArrayValue {
+    repeated google.protobuf.UInt64Value array = 1;   
+}
+	
+// TODO: what do you think we don't have a uint16 type in protobuf,
+// so we have a message to inform the size, so the user converts the internal value
+message UInt16Value {
+    uint32 value = 1; 
+}
+
+// TODO: what do you think we don't have a uint16 type in protobuf,
+// so we have a message to inform the size, so the user converts the internal value
+message UInt8Value {
+    uint32 value = 1; 
+}
+	
+message CredValue {
+    google.protobuf.UInt32Value uid = 1;
+    google.protobuf.UInt32Value gid = 2;
+    google.protobuf.UInt32Value suid = 3;
+    google.protobuf.UInt32Value sgid = 4;
+    google.protobuf.UInt32Value euid = 5;
+    google.protobuf.UInt32Value egid = 6;
+    google.protobuf.UInt32Value fsuid = 7;
+    google.protobuf.UInt32Value fsgid = 8;
+    google.protobuf.UInt32Value user_namespace = 9;
+    google.protobuf.UInt32Value secure_bits = 10;
+    repeated Capability cap_inheritable = 11;
+    repeated Capability cap_permitted = 12;
+    repeated Capability cap_effective = 13;
+    repeated Capability cap_bounding = 14;
+    repeated Capability cap_ambient = 15;
+}
+	
+message TimespecValue {
+    google.protobuf.DoubleValue value = 1;
+}
+
+// TODO: the syscall args type will have an array of event data?
```

Comment:
Yes, but again with the exception it won't have TriggerEvent and SyscallArgsValue as one of its types.